src/main/java/pl/piekoszek/gorskimatches/equation/EquationMathChecker.java

```diff
@@ -2,11 +2,36 @@
 
 import org.springframework.stereotype.Component;
 
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
 @Component
 class EquationMathChecker {
 
+    Map<String, Boolean> allEquations = new HashMap<>();
+    ArrayList equations = new ArrayList();
+
+    {
+        for (int i = 0; i < 10; i++) {
+            for (int j = 0; j < 10; j++) {
+                for (int k = 0; k < 10; k++) {
+                    equations.add(i + "+" + j + "=" + k);
+                    equations.add(i + "-" + j + "=" + k);
+                    for (int l = 0; l < equations.size(); l++) {
+                        allEquations.put((String) equations.get(l), isMathematicallyCorrect((String) equations.get(l)));
+                        allEquations.remove(equations.get(l), false);
```

Comment:
Good job finding this method :D

Other option would be:
```
for (String equation : equations) {
    if( isMathematicallyCorrect(equation)) {
        listOfMathCorrEquations.add(equation);
        }
    }
}
```

in that case, you wouldn't need Map at all